test/irb/test_cmd.rb

```diff
@@ -469,6 +469,116 @@ def test_show_source_method
       assert_match(%r[/irb\/init\.rb], out)
     end
 
+    def test_show_source_method_s
+      code = <<~RUBY
+        class Baz
+          def foo
+          end
+        end
+      
+        class Bar < Baz
+          def foo
+            super
+          end
+        end
+      RUBY
+      File.write("#{@tmpdir}/bazbar.rb", code)
+      out, err = execute_lines(
+        "irb_load '#{@tmpdir}/bazbar.rb'\n",
+        "show_source Bar#foo -s",
+      )
+      assert_match(%r[bazbar.rb:2\n\n  def foo\n  end\n\n=> nil\n], out)
+    end
+
+    def test_show_source_method_multiple_s
+      code = <<~RUBY
+        class Baz
+          def foo
+          end
+        end
+      
+        class Bar < Baz
+          def foo
+            super
+          end
+        end
+
+        class Bob < Bar
+          def foo
+            super
+          end
+        end
+      RUBY
+      File.write("#{@tmpdir}/bazbarbob.rb", code)
+      out, err = execute_lines(
+        "irb_load '#{@tmpdir}/bazbarbob.rb'\n",
+        "show_source Bob#foo -ss",
+      )
+      assert_match(%r[bazbarbob.rb:2\n\n  def foo\n  end\n\n=> nil\n], out)
+    end
+
+    def test_show_source_method_no_instance_method
+      code = <<~RUBY
+        class Baz
+        end
+      
+        class Bar < Baz
+          def foo
+            super
+          end
+        end
+      RUBY
+      File.write("#{@tmpdir}/bazbar.rb", code)
+      out, err = execute_lines(
+        "irb_load '#{@tmpdir}/bazbar.rb'\n",
+        "show_source Bar#foo -s",
+      )
+      assert_match(%r[Error: Couldn't locate a super definition for Bar#foo\n], out)
```

Comment:
CI (ruby-core setup) is failing in this line.
You can reproduce it by removing `def test_show_source_method_multiple_s` and run `rake test TEST=test/irb/test_cmd.rb` in local.

The problem is that after other test, `Object::Baz` still remains with method `foo` defined.
In this case, `class Baz; end` is not a class definition but an open-class.
The test accidentally passed in other CI because source_location file `bazbarbob.rb` was already deleted.

I think the easiest way to fix this is to add these in all added tests
```ruby
ensure
  Object.remove_const :Bar
  Object.remove_const :Baz
  Object.remove_const :Bob
end
```

Reply:
I ended up just changing the method names in each test since it wouldn't let me access the `remove_const` method now these pass even when commenting out `def test_show_source_method_multiple_s`

TruffleRuby CI is failing because of a gem I believe:
```ruby
In Gemfile:
  debug was resolved to 1.8.0, which depends on
    irb was resolved to 1.9.1, which depends on
      rdoc was resolved to 6.6.0, which depends on
        psych was resolved to 5.1.1.1, which depends on
          stringio
Error: The process '/home/runner/.rubies/truffleruby-head/bin/bundle' failed with exit code 5
```

Reply:
(Don't worry about truffleruby for now, we're aware of the problem and have contacted related teams)